Add X-Source header to Gravatar REST API request

hamorillo opened the pull request:
Closes #360 

### Description

This is a follow-up of #353. 

Having the request's source can help understand how developers use the SDK and their needs.

If the app provides the SDK with the context through the `Gravatar` object, we'll use that context to extract the application's name and send it in the `X-Source` header. 

Currently, we only use the `Context` for that purpose; it's optional to provide it. When it's not provided, we won't send the header.

### Testing Steps

1. Navigate to the QE tab in the demo-app
2. Switch through different avatars (Any QE request should work).
3. With the `Network Inspector`, verify you can see the header
<img width="325" alt="image" src="https://github.com/user-attachments/assets/138a94a4-3883-4a7e-91fa-ebe472d76670">

4. In the demo-app code, remove the context from the initialization and run the demo-app
```
// Initialize the Gravatar SDK with the API key if it is available
BuildConfig.DEMO_GRAVATAR_API_KEY?.let { Gravatar.apiKey(it) }
``` 
5. Navigate to the QE tab in the demo-app
6. Switch through different avatars
7. With the `Network Inspector`, verify you CAN'T see the header
8. Everything else works as expected.

hamorillo committed:
Accept Context when initializating the Gravatar SDK

If the context is provided, we will extract the app name to be used in the request headers.

hamorillo committed:
Add X-Source in the SdkVersionInterceptor

When the appName is available, we should send the X-Source header with that value.

hamorillo committed:
Update documentation

AdamGrzybkowski commented on demo-app/src/main/java/com/gravatar/demoapp/MainActivity.kt:
```diff
@@ -13,7 +13,7 @@ class MainActivity : ComponentActivity() {
 
         // Initialize the Gravatar SDK with the API key if it is available
-        BuildConfig.DEMO_GRAVATAR_API_KEY?.let { Gravatar.initialize(it) }
+        BuildConfig.DEMO_GRAVATAR_API_KEY?.let { Gravatar.apiKey(it).context(this) }
```
```suggestion
        BuildConfig.DEMO_GRAVATAR_API_KEY?.let { Gravatar.apiKey(it).context(applicationContext) }
```

I guess it works, but I think this should be the application context, no?

hamorillo replied:
I would say it should work with any context. But let's change it to prevent issues.

AdamGrzybkowski commented on docs/get-started/get-started.md:
````diff
@@ -63,7 +63,7 @@ android {
 
 ```kotlin
-Gravatar.initialize(BuildConfig.GRAVATAR_API_KEY)
+Gravatar.apiKey(BuildConfig.GRAVATAR_API_KEY).context(appContext)
````
```suggestion
Gravatar.apiKey(BuildConfig.GRAVATAR_API_KEY)
    .context(appContext) // Optional
```

Should we add a comment that the `context` is optional?

hamorillo replied:
I would say it should work with any context. But let's change it to prevent issues.

hamorillo replied:
🤔 What about: 

```
// Optional but highly encouraged.
``` 

As it provides valuable information for us.

AdamGrzybkowski replied:
👍

AdamGrzybkowski reviewed (approved):
Works as described 👍 Left two minor comments.

hamorillo committed:
Update demo-app/src/main/java/com/gravatar/demoapp/MainActivity.kt

I would say it should work with any context. But let's change it to prevent issues.

Co-authored-by: Adam Grzybkowski <agrzybkowski@outlook.com>

hamorillo committed:
Improve get-started
